webextension/scripts/settings.js

```diff
@@ -79,6 +79,15 @@ function saveOptions() {
     chrome.runtime.sendMessage({ message: 'clearCountBadge' })
     chrome.runtime.sendMessage({ message: 'clearCountCache' })
   }
+  if (wiki_resource === false) {
+    chrome.runtime.sendMessage({ message: 'clearResource', resource: 'wikiResource' })
+  }
+  if (amazon_books === false) {
+    chrome.runtime.sendMessage({ message: 'clearResource', resource: 'amazonBooks' })
+  }
+  if (tv_news === false) {
+    chrome.runtime.sendMessage({ message: 'clearResource', resource: 'tvNews' })
+  }
```

Comment:
@cgorringe, I made these changes here to clear the toolbar icon for specific resource.